internal/k8s/spiffe.go

```diff
@@ -10,20 +10,21 @@ import (
 	"github.com/spiffe/go-spiffe/workload"
 )
 
-type spiffeController struct {
+// SpiffeController controls spiffe
```

Comment:
Unrelated changes?

Reply:
No unfortunately the linter companies about many files, so this got fixed as part of trying to stop the linter from failing.

Reply:
it shouldn't complain about files that weren't modified

Reply:
Well you see, when the linting fails, it gives a very unhelpful message. i.e. `file is not gofumpt -ed`. This is in contrast to normal linters that say things like `dos.go:123 function is missing a space before curly bracket`. Also, as you can see here, https://github.com/nginxinc/kubernetes-ingress/runs/4507864465?check_suite_focus=true it doesn't show you which file is not `gofumpt -ed`. 

This means the developer needs to run the linter locally. Unfortunately, the linter reports many, many issues. That's where we get to why `SpiffeController` was updated. `make lint` actually reports all linting issues but it seems the listing action on GitHub only fails for certain linters. If I run it locally this morning I see several errors:
```
tests/test-servers/tcp/main.go:44:19: Error return value of `conn.Close` is not checked (errcheck)
		defer conn.Close()
		                ^
internal/metrics/collectors/processes.go:60:19: G304: Potential file inclusion via variable (gosec)
		content, err := ioutil.ReadFile(cmdlineFile)
```
But interestingly these don't turn up in the Github lint action!
These means when I was trying to fix the linting errors, I didn't know which ones I need to fix. So I started fixing them all. Hence, the `SpiffeController` now has an additional comment.

Also, gofumpt and gofmt are not dependencies of the project. This means a developer needs to independently download those tools, figure out their arguments and run them manually.

Also, since if we touch a file, it must be perfectly linted, it places a large burden on the developer. This PR added code to `ParseConfigMap` which causes the linter to fail as it's cyclomatic complexity of 135 exceeds the 15 threshold. I should not have to pause development and refactor an unrelated 500 line function to get the DOS feature onto the main branch.